notify/silences.go

```diff
@@ -124,3 +129,54 @@ func (am *GrafanaAlertmanager) DeleteSilence(silenceID string) error {
 
 	return nil
 }
+
+func (am *GrafanaAlertmanager) SilenceState() (SilenceState, error) {
+	r, w := io.Pipe()
```

Comment:
There are a couple of things here that I don't understand:

1. How come we snapshot the silence in a goroutine?
2. What's the reason behind using `io.Pipe` instead of `bytes.Buffer`?

Reply:
If `io.Pipe` is more performant then I'm 👍

Reply:
`io.Pipe()` uses channels for both the reader and the writer, so calling `am.silences.Snapshot(w)` in the same goroutine would block indefinitely.